services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/QueryEnhancedRequest.java

```diff
@@ -255,6 +273,40 @@ public Builder filterExpression(Expression filterExpression) {
             return this;
         }
 
+        /**
+         * <p>
+         * A string that identifies one or more attributes to retrieve from the table. These attributes can include
+         * scalars, sets, or elements of a JSON document. The attributes in the expression must be separated by commas.
+         * </p>
+         * <p>
+         * If no attribute names are specified, then all attributes will be returned. If any of the requested attributes
+         * are not found, they will not appear in the result.
+         * </p>
+         * <p>
+         * For more information, see <a href=
+         * "https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.AccessingItemAttributes.html"
+         * >Accessing Item Attributes</a> in the <i>Amazon DynamoDB Developer Guide</i>.
+         * </p>
+         *
+         * @param projectionExpression
+         *        A string that identifies one or more attributes to retrieve from the table. These attributes can
+         *        include scalars, sets, or elements of a JSON document. The attributes in the expression must be
+         *        separated by commas.</p>
+         *        <p>
+         *        If no attribute names are specified, then all attributes will be returned. If any of the requested
+         *        attributes are not found, they will not appear in the result.
+         *        </p>
+         *        <p>
+         *        For more information, see <a href=
+         *        "https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.AccessingItemAttributes.html"
+         *        >Accessing Item Attributes</a> in the <i>Amazon DynamoDB Developer Guide</i>.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         */
+        public Builder projectionExpression(String projectionExpression) {
```

Comment:
I'm thinking that since this is a high-level client, we should model this as '''List<String> attributesToProject''' or something like that? What do you think?